### PR TITLE
Select text on focus and autoselect option on typed

### PR DESCRIPTION
**Pull request by anandphulwani**
- Created plugin selecttext_onfocus_and_autoselectoption_on_typed which does the following:
   - Select text on the input when focused in.
   - Highlight option in dropdown whenever focused into the component.
   - Auto select option when option is typed in input box and focused out.
 - Added functions like getTextboxValue, getElementWithTextContent, getFirstItemMatchedByTextContent
   - **getTextboxValue**: Gets the current value of input control.
   - **getElementWithTextContent**: Finds the first element with a "textContent" property (innerHTML) that matches the given string provided for search..
   - **getFirstItemMatchedByTextContent**: Returns the jQuery element of the item matching the given textContent / innerHTML.

This should resolve the issue mentioned here https://github.com/selectize/selectize.js/issues/996

To use just do 
```
element.selectize({
    plugins: ['selecttext_onfocus_and_autoselectoption_on_typed']
});
```

### CODE REVIEW

File: src/plugins/selecttext_onfocus_and_autoselectoption_on_typed/plugin.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+Selectize.define('selecttext_onfocus_and_autoselectoption_on_typed', function(options) {
```
Comment (risadams) [inline src/plugins/selecttext_onfocus_and_autoselectoption_on_typed/plugin.js]: Would this make more sense as two separate plugins?

**Review by risadams — CHANGES_REQUESTED** [review]
Can you please try to add a sample to the /examples/plugins.html file to help with testing?
I think that the plugin name itself is quite long, and should probably be split into two separate plugins if possible.

Thanks

**Comment by anandphulwani** [comment]
> 
> 
> Can you please try to add a sample to the /examples/plugins.html file to help with testing?
> I think that the plugin name itself is quite long, and should probably be split into two separate plugins if possible.
> 
> Thanks

 - Yes i can add samples to the /examples/plugins.html file, will do it.
 - We can split into two separate plugins, but it would require the 3 getter methods, to be integrated in the selectize.js file first, so 
if you want that, i will close this PR and send a single PR for the getter methods first, and the two plugins PR afterwards, you advise.

**Comment by risadams** [comment]
> * We can split into two separate plugins, but it would require the 3 getter methods, to be integrated in the selectize.js file first, so
>   if you want that, i will close this PR and send a single PR for the getter methods first, and the two plugins PR afterwards, you advise.

I'm not sure if I follow,  you should be able to have all changes in the same branch, and in a single PR

File: src/plugins/selecttext_onfocus_and_autoselectoption_on_typed/plugin.js (hunk 1 above)
Comment (anandphulwani) [reply src/plugins/selecttext_onfocus_and_autoselectoption_on_typed/plugin.js]: Done.

**Comment by anandphulwani** [comment]
Changes have been done. plugins can be used together and independently as well by
```
element.selectize({
    plugins: ['selecttext_on_focus'],
});
```
and
```
element.selectize({
    plugins: ['autoselectoption_on_typed'],
});
```

**Review by risadams — APPROVED** [review]
I'll merge this now, but will probably modify the plugin names to be a bit more clear to understand. I'll talk you to you about adding some examples next.

Thanks

**Review by risadams — APPROVED** [review]
I'll merge this now, but will probably modify the plugin names to be a bit more clear to understand. I'll talk you to you about adding some examples next.

Thanks

**Review by risadams — APPROVED** [review]
I'll merge this now, but will probably modify the plugin names to be a bit more clear to understand. I'll talk you to you about adding some examples next.

Thanks

**Review by risadams — APPROVED** [review]
I'll merge this now, but will probably modify the plugin names to be a bit more clear to understand. I'll talk you to you about adding some examples next.

Thanks

**Comment by anandphulwani** [comment]
No issues, i would need some help for examples and tests to be created.

**Pull request merged**